server/src/main/java/org/opensearch/search/pipeline/ProcessorExecutionDetail.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline;
+
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.common.xcontent.XContentUtils;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * Detailed information about a processor execution in a search pipeline.
+ *
+ * @opensearch.internal
+ */
+@PublicApi(since = "2.19.0")
+public class ProcessorExecutionDetail implements Writeable, ToXContentObject {
+
+    private final String processorName;
+    private long durationMillis;
+    private Object inputData;
+    private Object outputData;
+    public static final ParseField PROCESSOR_NAME_FIELD = new ParseField("processor_name");
+    public static final ParseField DURATION_MILLIS_FIELD = new ParseField("duration_millis");
+    public static final ParseField INPUT_DATA_FIELD = new ParseField("input_data");
+    public static final ParseField OUTPUT_DATA_FIELD = new ParseField("output_data");
```

Comment:
All the above vars can be changed to `Private` since we are not using them anywhere apart from this class